src/main/scala/mesosphere/marathon/core/instance/Reservation.scala

```diff
@@ -12,6 +12,34 @@ import play.api.libs.json._
 case class Reservation(volumeIds: Seq[LocalVolumeId], state: Reservation.State)
 
 object Reservation {
+
+  /**
+    * Reservation.Id represents a reservation ID which is used in a task reservation labels.
+    */
+  case class Id(runSpecId: String, separator: String, prefix: Option[String], uuid: String) {
+    override def toString: String = prefix match {
+      case Some(prefixStr) => runSpecId + separator + prefixStr + uuid
+      case None => runSpecId + separator + uuid
+    }
+
+    def instanceId: Instance.Id = prefix match {
+      case Some(prefixStr) => Instance.Id(runSpecId + separator + prefixStr + uuid)
+      case None => Instance.Id(runSpecId + separator + "marathon-" + uuid)
+    }
+  }
+
+  object Id {
+    private val ReservationIdWithInstanceIdRegex = """^(.+)\.(instance-|marathon-)([^_\.]+)$""".r
```

Comment:
Why not reuse the pattern from `Task.Id`?

Reply:
It is different here.